packages/ui/src/types/index.ts

```diff
@@ -1,2 +1,6 @@
 export * from './authMachine';
 export * from './validator';
+
+// Prevents usage of T from being automatically inferred.
+// https://github.com/Microsoft/TypeScript/issues/14829#issuecomment-504042546
+export type NoInfer<T> = [T][T extends any ? 0 : never];
```

Comment:
This prevents premature template inference from Typescript.

This ensures that `translate('I'm a custom or dynamic string!')` is not be accepted, but `translate<string>('I'm a custom or dynamic string!')` is, so that we still enjoy strong authenticator string typings.

Reply:
That's clever! 👍